File: Adapter/Cpp/SquarePegAdapter.cpp
#include "SquarePegAdapter.h"
#include "RoundPeg.h"

SquarePegAdapter::SquarePegAdapter(SquarePeg inSquarePeg) :RoundPeg(inSquarePeg.Radius()), squarePeg(inSquarePeg)
{
}

File: Adapter/CSharp/SquarePegAdapter.cs
using System;

namespace CSharp
{
  class SquarePegAdapter : RoundPeg
  {
    //SquarePeg _squarePeg;

    public SquarePegAdapter( SquarePeg squarePeg ) : base( squarePeg.Width / Math.Sqrt( 2.0 ) )
    {
      //_squarePeg = squarePeg;
    }

  }
}
